http://sourceware.org/ml/gdb-cvs/2010-12/msg00071.html

### src/gdb/ChangeLog	2010/12/14 21:07:48	1.12382
### src/gdb/ChangeLog	2010/12/15 17:27:56	1.12383
## -1,3 +1,13 @@
+2010-12-15  Doug Evans  <dje@google.com>
+
+	Rename "maint set python auto-load" to "set auto-load-scripts".
+	* NEWS: Mention it.
+	* python/py-auto-load.c (auto_load_scripts): Renamed from
+	gdbpy_auto_load.
+	(load_auto_scripts_for_objfile): Update.
+	(gdbpy_initialize_auto_load): Rename "maint set python auto-load" to
+	"set auto-load-scripts".
+
 2010-12-14  Kevin Buettner  <kevinb@redhat.com>
 
 	* mips-tdep.c (mips_eabi_push_dummy_call): Place signed, rather
Index: gdb-7.2/gdb/NEWS
===================================================================
--- gdb-7.2.orig/gdb/NEWS
+++ gdb-7.2/gdb/NEWS
@@ -28,6 +28,9 @@
   libthread_db library with the "set libthread-db-search-path"
   command.  See the user manual for more details on this command.
 
+* The "maint set python auto-load on|off" command has been renamed to
+  "set auto-load-scripts on|off".
+
 *** Changes in GDB 7.2
 
 * Shared library support for remote targets by default
Index: gdb-7.2/gdb/python/py-auto-load.c
===================================================================
--- gdb-7.2.orig/gdb/python/py-auto-load.c
+++ gdb-7.2/gdb/python/py-auto-load.c
@@ -34,7 +34,7 @@
    This is true if we should auto-load python code when an objfile is opened,
    false otherwise.
 
-   Both gdbpy_auto_load && gdbpy_global_auto_load must be true to enable
+   Both auto_load_scripts && gdbpy_global_auto_load must be true to enable
    auto-loading.
 
    This flag exists to facilitate deferring auto-loading during start-up
@@ -82,10 +82,15 @@ struct loaded_script_entry
 };
 
 /* User-settable option to enable/disable auto-loading:
-   maint set python auto-load on|off
-   This is true if we should auto-load python code when an objfile is opened,
-   false otherwise.  */
-static int gdbpy_auto_load = 1;
+   set auto-load-scripts on|off
+   This is true if we should auto-load associated scripts when an objfile
+   is opened, false otherwise.
+   At the moment, this only affects python scripts, but there's no reason
+   one couldn't also have other kinds of auto-loaded scripts, and there's
+   no reason to have them each controlled by a separate flag.
+   So we elide "python" from the name here and in the option.
+   The fact that it lives here is just an implementation detail.  */
+static int auto_load_scripts = 1;
 
 /* Per-program-space data key.  */
 static const struct program_space_data *auto_load_pspace_data;
@@ -406,7 +411,7 @@ auto_load_new_objfile (struct objfile *o
 void
 load_auto_scripts_for_objfile (struct objfile *objfile)
 {
-  if (gdbpy_auto_load && gdbpy_global_auto_load)
+  if (auto_load_scripts && gdbpy_global_auto_load)
     {
       auto_load_objfile_script (objfile, GDBPY_AUTO_FILE_NAME);
       auto_load_section_scripts (objfile, GDBPY_AUTO_SECTION_NAME);
@@ -473,14 +478,15 @@ gdbpy_initialize_auto_load (void)
 
   observer_attach_new_objfile (auto_load_new_objfile);
 
-  add_setshow_boolean_cmd ("auto-load", class_maintenance,
-			   &gdbpy_auto_load, _("\
-Enable or disable auto-loading of Python code when an object is opened."), _("\
-Show whether Python code will be auto-loaded when an object is opened."), _("\
-Enables or disables auto-loading of Python code when an object is opened."),
+  add_setshow_boolean_cmd ("auto-load-scripts", class_support,
+			   &auto_load_scripts, _("\
+Set the debugger's behaviour regarding auto-loaded scripts."), _("\
+Show the debugger's behaviour regarding auto-loaded scripts."), _("\
+If enabled, auto-loaded scripts are loaded when the debugger reads\n\
+an executable or shared library."),
 			   NULL, NULL,
-			   &set_python_list,
-			   &show_python_list);
+			   &setlist,
+			   &showlist);
 
   add_cmd ("section-scripts", class_maintenance, maintenance_print_section_scripts,
 	   _("Print dump of auto-loaded section scripts matching REGEXP."),
Index: gdb-7.2/gdb/doc/gdb.texinfo
===================================================================
--- gdb-7.2.orig/gdb/doc/gdb.texinfo
+++ gdb-7.2/gdb/doc/gdb.texinfo
@@ -1264,6 +1264,25 @@ to the program you are debugging, in the
 @value{GDBN}.
 
 @item
+If the command line specified a program to debug, or a process to
+attach to, or a core file, @value{GDBN} loads any auto-loaded
+scripts provided for the program or for its loaded shared libraries.
+@xref{Auto-loading}.
+
+If you wish to disable the auto-loading during startup,
+you must do something like the following:
+
+@smallexample
+$ gdb -ex "set auto-load-scripts off" -ex "file myprogram"
+@end smallexample
+
+The following does not work because the auto-loading is turned off too late:
+
+@smallexample
+$ gdb -ex "set auto-load-scripts off" myprogram
+@end smallexample
+
+@item
 Reads command files specified by the @samp{-x} option.  @xref{Command
 Files}, for more details about @value{GDBN} command files.
 
@@ -22785,13 +22804,13 @@ debugging commands and scripts.
 Auto-loading can be enabled or disabled.
 
 @table @code
-@kindex maint set python auto-load
-@item maint set python auto-load [yes|no]
-Enable or disable the Python auto-loading feature.
-
-@kindex maint show python auto-load
-@item maint show python auto-load
-Show whether Python auto-loading is enabled or disabled.
+@kindex set auto-load-scripts
+@item set auto-load-scripts [yes|no]
+Enable or disable the auto-loading of Python scripts.
+
+@kindex show auto-load-scripts
+@item show auto-load-scripts
+Show whether auto-loading of Python scripts is enabled or disabled.
 @end table
 
 When reading an auto-loaded file, @value{GDBN} sets the
